packages/clerk-js/src/core/resources/Passkey.ts

```diff
@@ -0,0 +1,147 @@
+import type { PasskeyJSON, PasskeyResource, PasskeyVerificationResource } from '@clerk/types';
+
+import { unixEpochToDate } from '../../utils/date';
+import {
+  bufferToBase64Url,
+  handlePublicKeyCreateError,
+  isWebAuthnPlatformAuthenticatorSupported,
+} from '../../utils/passkeys';
+import { BaseResource, ClerkRuntimeError, PasskeyVerification } from './internal';
+
+export class Passkey extends BaseResource implements PasskeyResource {
+  id!: string;
+  pathRoot = '/me/passkeys';
+  credentialId: string | null = null;
+  verification: PasskeyVerificationResource | null = null;
+  name: string | null = null;
+  lastUsedAt: Date | null = null;
+  createdAt!: Date;
+  updatedAt!: Date;
+
+  public constructor(data: PasskeyJSON) {
+    super();
+    this.fromJSON(data);
+  }
+
+  private static async startRegistration() {
+    return await BaseResource._fetch({
+      path: `/me/passkeys`,
+      method: 'POST',
+    }).then(res => new Passkey(res?.response as PasskeyJSON));
+  }
+
+  private static async finishRegistration(publicKeyCredential: any) {
+    const headers = new Headers({
+      'Content-Type': 'application/json',
+    });
```

Comment:
We can leave it as the default `application/x-www-form-urlencoded` and try to send the credential payload under a param `public_key_credential`

Reply:
Yes, thanks for the reminder